src/strands/agent/agent.py

```diff
@@ -469,23 +474,21 @@ async def stream_async(self, prompt: str, **kwargs: Any) -> AsyncIterator[Any]:
 
         try:
             events = self._run_loop(callback_handler, prompt, kwargs)
-            for event in events:
+            async for event in events:
                 if "callback" in event:
                     callback_handler(**event["callback"])
                     yield event["callback"]
 
-            stop_reason, message, metrics, state = event["stop"]
-            result = AgentResult(stop_reason, message, metrics, state)
-
+            result = AgentResult(*event["stop"])
```

Comment:
💭 I wonder if the Async Iterator should be emitting something like this anyway (allowing us to unify the `__call__` method even more)